meerk40t/gui/propertypanels/imageproperty.py

```diff
@@ -131,10 +131,10 @@ def set_widgets(self, node=None):
         self.text_dpi.SetValue(str(node.dpi))
         try:
             bounds = node.bounds
-            self.text_x.SetValue(str(bounds[0]))
-            self.text_y.SetValue(str(bounds[1]))
-            self.text_width.SetValue(str((bounds[2] - bounds[0])))
-            self.text_height.SetValue(str((bounds[3] - bounds[1])))
+            self.text_x.SetValue("%.2fmm" % Length(amount=bounds[0], unitless=1).mm)
+            self.text_y.SetValue("%.2fmm" % Length(amount=bounds[1], unitless=1).mm)
+            self.text_width.SetValue("%.2fmm" % Length(amount=bounds[2]-bounds[0], unitless=1).mm)
+            self.text_height.SetValue("%.2fmm" % Length(amount=bounds[3]-bounds[1], unitless=1).mm)
```

Comment:
There's a `.length_mm` attribute that'll just show the length in right value. You can also set a limit to the number of decimals to consider significant in the Length() call there.

Reply:
Amended